test/hbbft_test_utils.erl

```diff
@@ -2,6 +2,13 @@
 
 -export([do_send_outer/4, shuffle/1, random_n/2, enumerate/1, merge_replies/3]).
 
+% TODO Type of Acc elements
+% TODO Type of States elements
+% TODO Type of Results elements
+-spec do_send_outer(Module :: atom(), Results :: list(), States, Acc) ->
```

Comment:
Specing a test-only function is kind of useless without some dialyzer contortions to run the dialyzer on test code (which can end up with a lot of noise because tests often intentionally call code with bad arguments).

Reply:
@Vagabond Make sense. Do you know how to do that through `rebar`? I see no obvious way other than adding a manual `dialyzer` call to the make recipe, like `dialyzer test/hbbft_test_utils.erl`

Reply:
You can configure the dialyzer to scan test code, somehow, but I don't think it is worth it.

Reply:
Gotcha. I was mainly thinking along the lines of documentation, since the spec is not immediately obvious. Alternatively, we could add it as a doc comment, but a spec is something that can at least potentially be checked, so a superior option, IMO.

Reply:
@Vagabond Added dialyzer to test recipe. Thanks, @evanmcc!